sdks/python/apache_beam/runners/direct/consumer_tracking_pipeline_visitor.py

```diff
@@ -65,5 +66,10 @@ def visit_transform(self, applied_ptransform):
       self.root_transforms.add(applied_ptransform)
     self.step_names[applied_ptransform] = 's%d' % (self._num_transforms)
     self._num_transforms += 1
+
     for side_input in applied_ptransform.side_inputs:
-      self.views.append(side_input)
+      if side_input not in self._side_input_views:
+        self._side_input_views[side_input] = side_input
+    self.views = [
```

Comment:
Here we may want to only append rather than rebuild all of `self.views`?

An idea is that we may want to keep `self._views = set()`, add views to the set, and add a `@property` that returns the set as a list? (no need to implement this way, but we do want to make sure that we are not dropping elements from 'views' that we appended in previous `visit_transform` calls